From 769387ba608d888cc6d629db48da749e90d1a1b7 Mon Sep 17 00:00:00 2001
From: Dinu Mathai <dinu.mathai@t-mobile.com>
Date: Mon, 17 Dec 2018 12:45:38 +0530
Subject: [PATCH 2/2] Bug Fix

(cherry picked from commit 8fae543dd9529af77bbe6f23ad84acbb1cb8bb30)
Signed-off-by: Tom Wieczorek <twieczorek@mirantis.com>
---
 README.md                  | 1 +
 storage/diskmetricstore.go | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/README.md b/README.md
index e4f13f1..ef05c0e 100644
--- a/README.md
+++ b/README.md
@@ -77,6 +77,7 @@ docker run -d -p 9091:9091 prom/pushgateway
 ### Time To Live
 If we pass a argument `metric.timetolive` at the time of start up(Example : `-metric.timetolive=60s`), 
 the metrics will be removed from pushgateway after the 'metric.timetolive' from the time of pushing the metric.
+**The clean up jobs runs every 60 sec. So anything below 60 sec will not work.** 
 
 ### Configure the Pushgateway as a target to scrape
 
diff --git a/storage/diskmetricstore.go b/storage/diskmetricstore.go
index a96114a..0d1ead3 100644
--- a/storage/diskmetricstore.go
+++ b/storage/diskmetricstore.go
@@ -464,7 +464,7 @@ func (dms *DiskMetricStore) doCleanUpInReguarInterval(timeToLive time.Duration)
 	}
 	for {
 		dms.cleanupStaleValues(timeToLive)
-		timer1 := time.NewTimer(timeToLive)
+		timer1 := time.NewTimer(60 * time.Second)
 		<-timer1.C
 	}
 }
@@ -486,6 +486,7 @@ func (dms *DiskMetricStore) cleanupStaleValues(timeToLive time.Duration) {
 		}
 	}
 }
+
 // groupingKeyFor creates a grouping key from the provided map of grouping
 // labels. The grouping key is created by joining all label names and values
 // together with model.SeparatorByte as a separator. The label names are sorted
-- 
2.36.0

